### DISCUSSION TITLE
Q&A

### DISCUSSION THREAD

**Discussion by PW25** [SAP-samples/integrated-business-planning-excel-addin-vba-samples#3] [Q&A]
Hi all, please share your questions right here.

**Comment by caspepet** [comment]
Hi all,
I am looking for a VBA capability to read certain master data values based on selected cells out of a planning view.
Just as an example to illustrate what I am looking for:
I double click on a cell which contains the product ID and the system should retrieve all product | location combinations which can be found in the master data table LOCATIONPRODUCT.
The available API "GetSingleMasterData" is only able to deliver attribute values for **one** fully qualified key.

**Reply by PW25** [reply]
Hi,
thank you very much for your request. Unfortunately for the next releases there are no plans to further enhance the VBA capabilities. But we add this item to our backlog and will review the request once we further improve the VBA capabilities.
Thanks
Patrick

**Comment by StefJen** [comment]
Hi,
I'm working on adding some custom VBA to our templates, but I'm occasionally getting this error message. If I refresh the view again then there's no problem. 
I've added some custom coding to AFTER_REFRESH() and IBPAfterRefresh() that will write to the name value of a local member and hide some columns. I'm using the way to set rptID from the samples:

`Dim rptID as String`
`rptID = IBPAutomationObject.GetActiveReportName(ActiveSheet)`

Anyone know how to avoid this error, or what could be causing it? Looking up in the documentation, knowledge base and googling have brought me nothing.
![image](https://github.com/SAP-samples/integrated-business-planning-excel-addin-vba-samples/assets/93737050/2db7fa45-93c5-40ac-8bdf-ad6b7aab77af)

**Reply by PW25** [reply]
Hi! Would you mind to open a ticket so that we have access to the sample and the affected system? Please attach your sample to the ticket or name the favorite/ template. Thanks Patrick Weber

**Reply by StefJen** [reply]
Hi Patrick.

I realized that the issue occurred because I copied the formulas from the key figure column to an array, made some modifications to certain entries, and then pasted the array back into the range of the key figure column. This seems to have disrupted the report's definition. Additionally, this caused another problem where data appeared twice if I had copied and pasted a range that was longer than the new selection in the view, resulting in the same combination being displayed multiple times.

So I've abandoned the idea again, and chosen another direction.

Best regards,
Steffen Jensen
